aws/graph-explorer
# Add Vitest to proxy server

## Description

Adds Vitest to the proxy server and updated some of the TypeScript setup, as well as a few dependency updates.

These are infrastructure tasks that are paving the way for the default connection logic coming in a future PR.

- Update TypeScript, ESLint, Prettier, Vitest to latest
  - Required fixing a few new warnings from Prettier and ESLint
- Add Vitest to proxy server
  - Add simple test of paths
  - Add coverage step for CI
- Move `check:types` script to each package
- Move proxy server source code files in to `./src/` directory
  - Updated relative path definitions and moved to a separate file for easier mocking
- Update root `package.json`
  - Add `private: true` to prevent accidental publishing
  - Add `workspaces` for slightly better monorepo functionality
  - Add `pnpm` to the `engines` section matching other popular monorepo setups

## Validation

- Smoke test production build
- Smoke test development build
- Run typical checks and tests

## Related Issues

- Part of #507 

### Check List



      license.
